src/odemis/util/graph.py

```diff
@@ -239,46 +218,320 @@ class WeightedGraph(GraphBase):
 
     _item_type = dict
 
-    def add_edge(self, edge: Tuple[int, int], weight: float = 1) -> None:
+    def add_edge(self, edge: Edge, weight) -> None:
```

Comment:
Any reason to remove the type of `weight`?

Reply:
I'd want to also use different types than `float` to use as `weight`. I believe we currently still need to support Python 3.7. Starting 3.8 it could be done using `typing.Protocol`:

```
class EdgeWeight(Protocol):
    """Defines a minimal interface for edge weights"""

    def __lt__(self, other: Any) -> bool:
        """Defines a total ordering"""

    def __float__(self) -> float:
        """Return the weight as a float"""
```